server/py/framework/db/base.py

```diff
@@ -236,6 +236,7 @@ def list_artifacts(
         producer_id: Optional[str] = None,
         producer_uri: Optional[str] = None,
         format_: mlrun.common.formatters.ArtifactFormat = mlrun.common.formatters.ArtifactFormat.full,
+        offset: Optional[int] = None,
         limit: Optional[int] = None,
```

Comment:
hmm... The existing `limit` field is actually already used by the UI to limit the total number of objects returned from the query. You cannot just change its meaning like this, since there are already existing APIs that use it.
I think the cleanest approach would be to rename these params - maybe just call it `page_offset` and `page_size`? This implies we're getting a single page starting at the offset and of the given size. WDYT?

Reply:
Spoke offline with @theSaarco and agreed on the necessary counter measures and we'll want to deprecate the `limit` in the API, but we'll support it for now.